docs/reference/api/app.rst

```diff
@@ -93,6 +97,60 @@ main window is closed, the app will exit.
 This is the type of app that will be created if you use an instance of :any:`toga.App`
 passing in a ``startup`` argument to the constructor.
 
+.. _session-based-app:
+
+Session-based app
+~~~~~~~~~~~~~~~~~
+
+A session-based app is an app that doesn't have a single main window. Instead, the app
+will have a number of windows, with each window corresponding to a "session" of
+interaction with the app. This session might be a editing a single document, or it could
+be a particular view of data - web browsers or file browsers would be examples of
+session-based apps, with each window representing a view of a URL, or a view of the file
+system.
+
+To define a session-based app, you assign a value of ``None`` as the main window. Your
+application code can then create new windows as required.
+
+The exact behavior of a session-based app is slightly different on each platform,
+reflecting platform differences.
+
+macOS
+^^^^^
+
+On macOS, there is only ever a single instance of an App running at any time. If you use
+the Finder to open a second document of a type managed by the app, it will be opened in
+the existing app instance. Closing all windows will not cause the app to exit; the app
+will keep executing until explicitly exited.
+
+If the app is started without an explicit file reference (e.g., by specifying a filename
+at the command line, or dragging a file onto the app's icon), a file dialog will be
+displayed prompting the user to select a file to open. If this dialog is dismissed, the
+app will continue running. Selecting "Open" from the file menu will also display this
+dialog. If a file is selected, a new document window will be opened.
+
+Linux/Windows
+^^^^^^^^^^^^^
+
+On Linux and Windows, a single app instance app can also manage multiple windows;
+however when the last window being managed by an app instance is closed, the app
+instance will exit. If the App is started without an explicit file reference, an empty
+document of the default document type will be opened.
+
+Mobile, web and console
+^^^^^^^^^^^^^^^^^^^^^^^
+
+Session-based apps are not supported on mobile, web and console platforms.
+
+Background app
+~~~~~~~~~~~~~~
+
+To create an app without *any* main window, assign ``toga.BACKGROUND`` as the main
+window. This will allow your app to persist even if it doesn't have any open windows. It
```

Comment:
`None` also means "an app without any main window".

```suggestion
If you set :attr:`~toga.App.main_window` to :attr:`App.BACKGROUND`, this will
allow your app to persist even if it doesn't have any open windows. It
```